### PR TITLE
Fixed #518 (Duplication of default favourite sites) for migration case

### LINKED ISSUE #518: Reorder favourites causes duplicates
### Description: 
Reorder favourites causes duplicates 

### Steps to Reproduce 
  1. Long press on favourite to edit mode
  2. Move one of them to a different position
  3. Favourite gets duplicated and doesn't come out of edit mode
  4. Tap on favourite loads the page but new tab still has them in edit mode
  5. Tap on edit button again and cancel brings them out of edit mode

**Actual result:** 
https://drive.google.com/open?id=1LFT5tgqpL5KJ6G4_vtaNL7YB1ETngJPJ

**Expected result:**
Should not duplicate and should come out of edit mode when tapped elsewhere

**Reproduces how often:** [Easily reproduced, Intermittent Issue]
Easy

**Brave Version:** 
1.7(18.11.30.3)

**Device details:** 
All

**Website problems only:**
- did you check with Brave Shields down?
- did you check in Safari/Firefox (WkWebView-based browsers)?

### PR DESCRIPTION
**Pull request by danishjafri88**
Fixes a case in #518 which happens during 1.6 to 1.7 migration. Since default favs are already created their syncOrder is nil during db migration. This causes duplication in Fav screen.

## Pull Request Checklist

- [ ] My patch has gone through review and I have addressed review comments
- [ ] My patch has a standard commit message that looks like `Fix #123: This fixes the shattered coffee cup!`

- [ ] I have updated the *Unit Tests* to cover new or changed functionality
- [ ] I have updated the *UI Tests* to cover new or changed functionality
- [ ] I have marked the bug with `[needsuplift]`
- [ ] I have made sure that localizable strings use `NSLocalizableString()`

## Screenshots

If your patch includes user interface changes that you would like to suggest or that you would like UX to look at, please include them here.

## Notes for testing this patch

If useful, please leave notes for QA, explaining what this patch changes and how it can be best tested and verified.

### CODE REVIEW

File: Client.xcodeproj/project.pbxproj
Comment (kylehickinson) [inline Client.xcodeproj/project.pbxproj]: Please remove all changes to `project.pbxproj`

File: Client/Entitlements/FennecApplication.entitlements
Comment (kylehickinson) [inline Client/Entitlements/FennecApplication.entitlements]: Please remove

**Review by kylehickinson — CHANGES_REQUESTED** [review]

File: Client.xcodeproj/project.pbxproj
Comment (danishjafri88) [reply Client.xcodeproj/project.pbxproj]: :) didnt uncheck these while committing. Removed

**Review by kylehickinson — DISMISSED** [review]

**Comment by kylehickinson** [comment]
Seems fine to me but will wait for @iccub to review as well before merging

**Comment by jhreis** [comment]
This may not be needed, so please do not merge yet.

**Comment by jhreis** [comment]
Thanks for this, we decided to revert back to using just the bookmark `order` property so this is not needed right now.

Refs:
https://github.com/brave/brave-ios/pull/602
https://github.com/brave/brave-ios/pull/600

**Pull request closed without merging**
